### ISSUE TITLE
Incorrect gas forwarding in DELEGATECALL definition

### ISSUE THREAD

**Issue by livnev** [ethereum/yellowpaper#732]
The definition of `DELEGATECALL` suggests that the amount of gas forwarded for execution is equal to the first stack argument, while in `CALL` the amount forwarded is given by the `C_callgas` formula. I'm pretty sure that the latter is correct, since the former would be kind of absurd, allowing arbitrary amounts of gas inside `DELEGATECALL` execution. To double check, the Jello Paper also supports the version where the amount forwarded is computed in the same way as in `CALL` (note the use of `Ccallgas`):

```

    syntax CallSixOp ::= "DELEGATECALL"
 // -----------------------------------
    rule <k> DELEGATECALL GCAP ACCTTO ARGSTART ARGWIDTH RETSTART RETWIDTH
          => #checkCall ACCTFROM 0
          ~> #call ACCTAPPFROM ACCTFROM ACCTTO Ccallgas(SCHED, #accountNonexistent(ACCTFROM), GCAP, GAVAIL, 0) 0 VALUE #range(LM, ARGSTART, ARGWIDTH) false
          ~> #return RETSTART RETWIDTH
         ...
         </k>
         <schedule> SCHED </schedule>
         <id> ACCTFROM </id>
         <caller> ACCTAPPFROM </caller>
         <callValue> VALUE </callValue>
         <localMem> LM </localMem>
         <previousGas> GAVAIL </previousGas>
```

So I believe the sixth argument of `Theta` in the definition of `DELEGATECALL` should be `C_callgas(mu)`. This is a critical issue since it means that implementers following the Yellow Paper would write incorrect semantics for `DELEGATECALL`.

**Comment by livnev** [comment]
Oh, it appears that the KEVM folks also noticed this issue: it's mentioned as the last bullet point in the section "Issues with description of EVM" [here](https://jellopaper.org/issues/#issues-with-description-of-evm).

**Comment by MrChico** [comment]
Ping @nicksavers

**Comment by nicksavers** [comment]
This has been part of the document since the addition of `DELEGATECALL` here https://github.com/ethereum/yellowpaper/commit/0de6d7d727b8eee85cf4801f0933c81e52238397#diff-9f702e1491c55da9d76a68d651278764R1981 in accordance with https://eips.ethereum.org/EIPS/eip-7
Currently checking with @aeronbuchanan & @gavofyork to verify that it was intentional or a mistake.

**Comment by livnev** [comment]
@nicksavers note that the easiest example that differentiates between what's written here and what is actually on the live network (and KEVM etc.) is: if you delegatecall with an amount of gas that's greater than the amount available, following what the yellow paper currently says literally tells you that you'll have that amount of gas available inside the delegatecall, i.e. there is no limit to how much gas you can have in the subcall. 

#### Explanation


![screenshot381](https://user-images.githubusercontent.com/3964494/55787994-6d3cf980-5aaf-11e9-99c6-fedc90f9055f.png)
![screenshot382](https://user-images.githubusercontent.com/3964494/55787997-6f9f5380-5aaf-11e9-9c15-78db759724b3.png)
![screenshot384](https://user-images.githubusercontent.com/3964494/55788152-b4c38580-5aaf-11e9-91a7-ed4b56c7a23e.png)
so if the element at the top of the stack when I delegatecall is `G`, the initial gas cost of the call will be `C_gascap = min(L(remaining_gas - 700), G)`, i.e. it's capped at `63/64 * (remaining_gas - 700)` no matter what. But meanwhile, the amount of gas available inside the delegatecall is given by sixth argument to `Theta` here:

![screenshot383](https://user-images.githubusercontent.com/3964494/55788002-70d08080-5aaf-11e9-9b53-f1efc6a69d8f.png)

i.e. `G` itself. Thus while our gas cost for the call is bounded, the amount of gas we are getting is unbounded, which is clearly absurd (and can be verified to be empirically not the case in all implementations). The proposed fix is in PR #733

**Comment by nicksavers** [comment]
Merged
